pkg/upstream/http.go

```diff
@@ -100,7 +105,35 @@ func newReverseProxy(target *url.URL, upstream options.Upstream, errorHandler Pr
 	if upstream.FlushInterval != nil {
 		proxy.FlushInterval = *upstream.FlushInterval
 	} else {
-		proxy.FlushInterval = 1 * time.Second
+		proxy.FlushInterval = defaultFlushInterval
+	}
+
+	var finalTimeout time.Duration
+	if upstream.Timeout != nil {
+		finalTimeout = *upstream.Timeout
+	} else {
+		finalTimeout = defaultTimeout
+	}
+
+	var finalKeepAlive time.Duration
+	if upstream.Timeout != nil {
+		finalKeepAlive = *upstream.KeepAlive
+	} else {
+		finalKeepAlive = defaultKeepAlive
+	}
+
+	// Ripped from http.DefaultTransport
+	proxy.Transport = &http.Transport{
```

Comment:
Since the `http.RoundTripper` implemented by `http.Transport` has all public fields -- is it potentially cleaner to just add in our updated `DialContext` settings rather than needing to track `http.DefaultTransport` and copying all the fields?

Reply:
Yeah this was just ignorance on my part. I wasn't sure how folks are copying/augmenting `http.DefaultTransport` when they want to modify the DialContext, I'll search the internet for examples tonight and implement them here.

Reply:
Turns out there's a "new" clone func: https://golang.org/pkg/net/http/#Transport.Clone